### PR TITLE
Add various functions

### PR DESCRIPTION
**Pull request by zorg93**

### CODE REVIEW

**Comment by Nacorpio** [comment]
Good job!

File: source/scripting/Game.cpp [hunk 1]
```diff
@@ -424,4 +424,20 @@ namespace GTA
...
 		return Native::Function::Call<bool>(Native::Hash::_GET_LAST_INPUT_METHOD, 2) ? InputMode::MouseAndKeyboard : InputMode::GamePad;
 	}
+	int Game::GetRandomInt(int max)
+	{
+		return GetRandomInt(0, max);
+	}
+	int Game::GetRandomInt(int min, int max)
+	{
+		return Native::Function::Call<int>(Native::Hash::GET_RANDOM_INT_IN_RANGE, min, max);
+	}
```
Comment (JohnnyCrazy) [inline source/scripting/Game.cpp]: Should we include Stuff like this?
IMO we bloat the code by implementing utility-functions which can be achieved by using simple C# default classes. Also, it has nothing to do with `Game`

@crosire

Comment (crosire) [inline source/scripting/Game.cpp]: Mmh. Yes. While it's a native, it's not really needed due to the .NET framework. I wouldn't add these too.

File: source/scripting/Player.hpp [hunk 2]
```diff
@@ -116,12 +117,26 @@ namespace GTA
 			int get();
 			void set(int value);
 		}
+		property Math::Vector3 WantedCentrePosition
```
Comment (crosire) [inline source/scripting/Player.hpp]: Should be "WantedCenterPosition", as we use American English everywhere else.

**Comment by zorg93** [comment]
Fixed those tweaks also added some parachute stuff

File: source/scripting/Player.cpp [hunk 3]
```diff
@@ -171,7 +173,52 @@ namespace GTA
...
+	}
+	void Player::WantedCenterPosition::set(Math::Vector3 value)
+	{
+		Native::Function::Call(Native::Hash::SET_PLAYER_WANTED_CENTRE_POSITION, Handle, value.X, value.Y, value.Z);
+	}
+	ParachuteTint Player::PrimaryParachuteTint::get()
+	{
+		int tint = 0;
+		Native::Function::Call(Native::Hash::GET_PLAYER_PARACHUTE_TINT_INDEX, Handle, &tint);
+		return static_cast<int>(tint);
```
Comment (crosire) [inline source/scripting/Player.cpp]: ```static_cast<ParachuteTint>(tint)``` =)

**Comment by zorg93** [comment]
Im not entirely sure what the wheel speed property is meant to return. but it matches up with the value on the cars speedo(when converted to mph) perfectly

File: source/scripting/Vehicle.hpp [hunk 4]
```diff
@@ -728,6 +728,14 @@ namespace GTA
 		{
 			float get();
 		}
+		/// <summary>
+		/// Gets the speed the wheels are spinning at
+		/// </summary>
+		/// <value>The speed in metres per second</value>
```
Comment (crosire) [inline source/scripting/Vehicle.hpp]: Not exactly important, but American English would be preferred, so we stay consistent =P (meters)

**Comment by crosire** [comment]
I'm not fully convinced we should use ```UI::ShowSubtitle``` to print an error in ```Ped::ChangeModel```. While it's convenient, no other class does it, they all print an error to the log file instead.

**Comment by zorg93** [comment]
Instead of logging the error. would return true if the change works/false if not be a better idea

**Comment by crosire** [comment]
Even better, yes.

**Comment by zorg93** [comment]
Removed Trunk2 see #463 for reason why

**Comment by zorg93** [comment]
Can you proof read the rest of the commit :)

File: source/scripting/Player.cpp [hunk 5]
```diff
@@ -105,6 +106,14 @@ namespace GTA
...
 		return Native::Function::Call<Vehicle ^>(Native::Hash::GET_PLAYERS_LAST_VEHICLE);
 	}
+	int Player::MaxArmour::get()
+	{
+		return Native::Function::Call<int>(Native::Hash::GET_PLAYER_MAX_ARMOUR, Handle);
+	}
+	void Player::MaxArmour::set(int value)
+	{
+		Native::Function::Call(Native::Hash::SET_PLAYER_MAX_ARMOUR, Handle, value);
+	}
```
Comment (JohnnyCrazy) [inline source/scripting/Player.cpp]: American english would be good, I know it's hard for the british people :stuck_out_tongue_winking_eye:

File: source/scripting/Player.hpp [hunk 6]
```diff
@@ -94,6 +112,11 @@ namespace GTA
 		{
 			Vehicle ^get();
 		}
+		property int MaxArmour
+		{
+			int get();
+			void set(int value);
+		}
 		property int Money
```
Comment (JohnnyCrazy) [inline source/scripting/Player.hpp]: ^Above

**Comment by JohnnyCrazy** [comment]
Besides one american --> english translation, all looks good to me :+1:

**Comment by zorg93** [comment]
Wait is it Armour or Armor, the native(which generally take american) says its armour

**Comment by kagikn** [comment]
Armor is American English, armour is British English.

**Comment by zorg93** [comment]
*armour is English :)

**Comment by crosire** [comment]
:smile:

**Comment by crosire** [comment]
Looks good now. I'll merge it later today.

**Comment by zorg93** [comment]
Do you want #466 being put in here or not, I have a feeling it could cause issues with people already using DrawTexture (textures may look off if scripts arent tweaked). I would also probably remove the overload that accepts the aspect ratio now the calc is done internally

**Comment by zorg93** [comment]
Credit To Guad for the subtitles and notifications, it was copied from NativeUI

**Comment by crosire** [comment]
@zorg93 I'd put that one in a separate pull request, so it can be further discussed.

File: source/scripting/Vehicle.cpp [hunk 7]
```diff
@@ -792,6 +800,18 @@ namespace GTA
 	{
 		return Native::Function::Call<System::String ^>(Native::Hash::GET_MOD_TEXT_LABEL, Handle, static_cast<int>(modType), modValue);
 	}
+	bool Vehicle::DoesExtraExist(int extra)
```
Comment (kagikn) [inline source/scripting/Vehicle.cpp]: I think Vehicle::ExtraExists would be better.

**Comment by zorg93** [comment]
Im certain that wheel speed property is correctly name now. The blue dial represents wheel speed, other is just entity speed
http://puu.sh/oWzB2/bb2ff66c18.jpg
http://puu.sh/oWzKr/17c930136a.jpg

File: source/scripting/Vehicle.cpp (hunk 7 above)
Comment (zorg93) [inline source/scripting/Vehicle.cpp]: The PR has been closed now, If you want it changed put in a new commit

**Pull request merged**